src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.ts

```diff
@@ -48,15 +48,19 @@ export class LoginComponent implements OnInit {
             window.location.replace(login.login_url);
           }
         } else {
-          this.authStorageService.set(login.username, token, login.permissions);
+          this.authStorageService.set(login.username, token, login.permission, 
+            Boolean(this.authService.getNeedChangePassword(login.username))); 
```

Comment:
If you define the method as returning `Boolean` you don't need to type-cast here.